### PR TITLE
onegoing interest

### PR DESCRIPTION
**Pull request by yuval-weiss**

### CODE REVIEW

File: contracts/PositionManagerOngoingInterest.sol [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+        (uint256 price,) = priceFeed.fetchPrice();
+        uint256 collateralBalance = raftCollateralToken.balanceOf(position);
+        uint256 debtBalance = debtToken.balanceOf(position);
+        uint256 MCR = IPositionManager(positionManager).splitLiquidationCollateral(wrappedCollateralToken).MCR();
+
+        uint256 maxDebt = collateralBalance * price / MCR;
+        return debtBalance >= maxDebt ? 0 : maxDebt - debtBalance;
+    }
+
+    function _applyBorrowingFee(uint256 amount)
```
Comment (yuval-weiss) [inline contracts/PositionManagerOngoingInterest.sol:153]: Alternatively, we can put the lock on pricefeed and disable redemptions completely. Then we can assume borrowing fee is always 0.

File: contracts/PositionManagerOngoingInterest.sol [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+import { MathUtils } from "./Dependencies/MathUtils.sol";
+import { IERC20WrappedLockable } from "./Interfaces/IERC20WrappedLockable.sol";
+import { IERC20Wrapped } from "./Interfaces/IERC20Wrapped.sol";
+import { IPositionManager } from "./Interfaces/IPositionManager.sol";
+import { IPriceFeed } from "./Interfaces/IPriceFeed.sol";
+import { PositionManagerWrappedCollateralToken } from "./PositionManagerWrappedCollateralToken.sol";
+import { PositionManagerDependent } from "./PositionManagerDependent.sol";
+
+/// TODO: IMPORTANT have this per collateral token or a global one? make it global probably
+contract PositionManagerOngoingInterest is Ownable2Step, PositionManagerWrappedCollateralToken {
```
Comment (StefanVitor) [inline contracts/PositionManagerOngoingInterest.sol:17]: Interface, documentation, override functions and variables...

File: contracts/Interfaces/IERC20WrappedLockable.sol [hunk 3]
```diff
@@ -0,0 +1,9 @@
+// SPDX-License-Identifier: BUSL-1.1
+pragma solidity 0.8.19;
+
+import { IWrappedCollateralToken } from "./IWrappedCollateralToken.sol";
+
+interface IERC20WrappedLockable is IWrappedCollateralToken {
```
Comment (mijovic) [inline contracts/Interfaces/IERC20WrappedLockable.sol:6]: you can now use the new Lock contract

Comment (yuval-weiss) [reply contracts/Interfaces/IERC20WrappedLockable.sol]: Done. Had to modify `Lock.sol` a bit.

**Pull request closed without merging**